src/garage/torch/algos/pearl_inference_network.py

```diff
@@ -0,0 +1,224 @@
+# pylint: disable=attribute-defined-outside-init
+"""An inference network that learns a latent probabilistic context variable Z.
+
+It is used in PEARL (Probabilistic Embeddings for Actor-Critic Reinforcement
+Learning). The paper on PEARL can be found at https://arxiv.org/abs/1903.08254.
+Code is adapted from https://github.com/katerakelly/oyster.
+"""
+
+import torch
+from torch import nn as nn
+import torch.nn.functional as F
+
+from garage.torch.utils import np_to_torch
+
+
+class PEARLInferenceNetwork(nn.Module):
```

Comment:
Usually the term inference network refers to the network that parameterizes the variational approximation to the posterior. I think it's weird to call something that contains the policy an inference network.

Reply:
How does inference agent sound? Or do you have any suggestions?